x-pack/test/detection_engine_api_integration/security_and_spaces/group1/export_rules.ts

```diff
@@ -49,8 +53,62 @@ export default ({ getService }: FtrProviderContext): void => {
           .expect('Content-Disposition', 'attachment; filename="export.ndjson"');
       });
 
-      it('should export a single rule with a rule_id', async () => {
-        await createRule(supertest, log, getSimpleRule());
+      it('should validate exported rule schema when its exported by its rule_id', async () => {
+        const ruleId = 'rule-1';
+
+        const rule = await createRule(supertest, log, getSimpleRule(ruleId, true));
+
+        await waitForRuleSuccessOrStatus(
+          supertest,
+          log,
+          rule.id,
+          RuleExecutionStatus['partial failure']
+        );
+        // to properly execute the test on rule's data with runtime fields some delay is needed as
+        // ES Search API may return outdated data
+        // it causes a reliable delay so exported rule's SO contains runtime fields returned via ES Search API
+        // and will be removed after addressing this issue
+        await waitForEventLogExecuteComplete(es, log, rule.id);
```

Comment:
To clarify, the event log and the `.kibana` index are separate. Just because the event log contains an "execution complete" event doesn't necessarily mean that the `.kibana` index has been updated and that the `execution_summary` has been added to the rule. It's possible for this update to happen before or after the execution event. We shouldn't rely on this assumption. 

The best option for us is to poll for the `execution_summary` update in the rule. But if for some reason, it is complicated, a better solution would be adding a "sleep" function for one second to ensure the `.kibana` index has been refreshed.

Reply:
Another thought is to try to await for `POST /.kibana/_refresh`.

Reply:
I've replaced `await waitForEventLogExecuteComplete` with 1 second delay.